Mathlib/MeasureTheory/Measure/FiniteMeasure.lean

```diff
@@ -762,12 +762,95 @@ theorem tendsto_iff_forall_integral_tendsto {γ : Type*} {F : Filter γ} {μs :
   exact Tendsto.sub tends_pos tends_neg
 #align measure_theory.finite_measure.tendsto_iff_forall_integral_tendsto MeasureTheory.FiniteMeasure.tendsto_iff_forall_integral_tendsto
 
-end FiniteMeasureConvergenceByBoundedContinuousFunctions
-
--- section
-end FiniteMeasure
-
--- namespace
-end MeasureTheory
-
--- namespace
+end FiniteMeasureConvergenceByBoundedContinuousFunctions -- section
+
+section map
+
+variable {Ω Ω' : Type _} [MeasurableSpace Ω] [MeasurableSpace Ω']
+
+/-- The push-forward of a finite measure by a function between measurable spaces. -/
+noncomputable def map (ν : FiniteMeasure Ω) (f : Ω → Ω') : FiniteMeasure Ω' :=
+  ⟨(ν : Measure Ω).map f, by
+    refine ⟨?_⟩
+    by_cases f_aemble : AEMeasurable f ν
+    · rw [Measure.map_apply_of_aemeasurable f_aemble MeasurableSet.univ]
+      exact measure_lt_top (↑ν) (f ⁻¹' univ)
+    · simp [Measure.map, f_aemble]⟩
+
+--#check Subtype.map
+-- Q: Can I tell Lean not to use `Subtype.map` but instead `FiniteMeasure.map`
+-- when `ν.map` is written?
```

Comment:
The problem here is that the current design allows us to build two different functions from `ν : MeasureTheory.FiniteMeasure` on `Set Ω`, one taking values in `ℝ≥0` and one in `ℝ≥0∞` BUT these do not form a definitionally-commuting triangle with the map `((↑) : ℝ≥0 → ℝ≥0∞)`. It's not immediately clear to me what the best solution for this is (one option would be to bundle an `ℝ≥0`-valued function with all measures) so I think I'd just add a comment here.
```suggestion
/-- Note that this is an equality of elements of `ℝ≥0∞`. See also
`MeasureTheory.FiniteMeasure.map_apply` for the corresponding equality as elements of `ℝ≥0`. -/
```